src/south/OPCHDA/OPCHDA.class.js

```diff
@@ -50,10 +54,17 @@ class OPCHDA extends ProtocolHandler {
   /**
    * On scan.
    * @param {String} scanMode - The scan mode
+   * @param {Date} startTime - The start time
+   * @param {Date} endTime - The end time
    * @return {Promise<void>} - The on scan promise
    */
-  onScanImplementation(scanMode) {
-    this.sendReadMessage(scanMode)
+  async historyQuery(scanMode, startTime, endTime) {
+    this.sendReadMessage(scanMode, startTime, endTime)
+    // Wait until we receive the response
+    while (this.ongoingReads[scanMode]) {
+      // eslint-disable-next-line no-await-in-loop
+      await this.delay(100)
```

Comment:
For `OPCUA-HA` and `SQLDbToFile` the `historyQuery` implementation returns a promise, so in `ProtolHandler` the `await historyQuery()` call will wait until the response is received. In case of `OPCHDA` the `sendReadMessage` is not returning a promise, so it will immediately return before we receive the response from the agent. The flag for `ongoingReads[scanMode]` is reset after we receive the response from the agent. The `while` loop waits until the response is received.

Reply:
I understand how this works but I would like to rethink this. I think we should await a promise and the response should resolve it. 
However, it's not really related to this PR so we could open a new issue for OPCHDA

Reply:
Something like this? https://stackoverflow.com/a/36072263/436312

Reply:
yes, seems like a good example. (but let's treat this as another issue)